### PR TITLE
print all known branch ancestry

### LINKED ISSUE #1299: git town config sometimes doesn't report parent relationships
Currently git town config only prints the trees under the main-branch / perennial branches and does not list parent configurations for branches that are children of a branch that does not have its parent configured.

Steps to reproduce:
* create branch `main1`, set as main branch 
* `git hack feature1`
* create branch `main2`, set as main branch
* `git town config` does not print the parent / child relationship between `main1` and `feature1`

### PR DESCRIPTION
**Pull request by charlierudolph**
resolves #1299

### CODE REVIEW

File: src/git/printable.go [hunk 1]
```diff
@@ -10,6 +10,22 @@ import (
...
 var noneString = "[none]"
 
+// getBranchAncestryRoots returns the branches with children and no parents
+func getBranchAncestryRoots() []string {
+	parentMap := Config().GetParentBranchMap()
+	roots := make([]string, len(parentMap))
+	i := 0
+	for _, parent := range parentMap {
+		if _, ok := parentMap[parent]; !ok && !util.DoesStringArrayContain(roots, parent) {
+			roots[i] = parent
```
Comment (kevgo) [inline src/git/printable.go:20]: You could simplify this by defining `roots` as `[]string{}`, and using `roots = append(roots, parent)` here. Then you don't need `i` and you don't need to truncate `roots` on line 24.

File: src/git/printable.go [hunk 2]
```diff
@@ -28,34 +44,34 @@ func GetPrintablePerennialBranches() string {
...
 	}
 	return
 }
 
+// GetPrintableBranchAncestry returns a user printable branch ancestry
+func GetPrintableBranchAncestry() string {
+	roots := getBranchAncestryRoots()
+	trees := make([]string, len(roots))
+	i := 0
+	for _, root := range roots {
```
Comment (kevgo) [inline src/git/printable.go:68]: `for i, root := range roots {` gives you the loop counter you need here.

File: src/git/printable.go [hunk 3]
```diff
@@ -10,6 +10,22 @@ import (
 
 var noneString = "[none]"
 
+// getBranchAncestryRoots returns the branches with children and no parents
+func getBranchAncestryRoots() []string {
```
Comment (kevgo) [inline src/git/printable.go:14]: If you want to unit-test logic like this, we have pretty good infrastructure for this. An example is https://github.com/git-town/git-town/blob/master/test/git_repository_test.go#L324-L342. `createTestGitTownRepo` creates a fully initialized Repo for you.

Comment (charlierudolph) [reply src/git/printable.go:14]: As time has passed going to skip this as feature test covers it well enough. Can add unit tests down the line if desired.

**Pull request merged**